shopgate.php

```diff
@@ -74,845 +147,624 @@ public function __construct()
-			`shop_number` varchar(16) NULL DEFAULT NULL,
-			`tracking_number` varchar(32) NOT NULL DEFAULT \'\',
-			`shipping_service` varchar(16) NOT NULL DEFAULT \'OTHER\',
-			`shipping_cost` decimal(17,2) NOT NULL DEFAULT \'0.00\',
-			`comments` text NULL DEFAULT NULL,
-			PRIMARY KEY (`id_shopgate_order`),
-			UNIQUE KEY `order_number` (`order_number`)
-		)
-		ENGINE=InnoDB DEFAULT CHARSET=utf8;';
-
-		$this->log('INSTALLATION - adding table "shopgate_order" to database', ShopgateLogger::LOGTYPE_DEBUG);
-		if (!$db->Execute($sql_table))
-		{
-			$this->log('installation failed: unable to add table "shopgate_order" to database. MySQL says: '.var_export($db->getMsgError(), true), ShopgateLogger::LOGTYPE_ERROR);
-			return false;
-		}
+		$db = Db::getInstance(true);
 
-		$this->log('installation after first query', ShopgateLogger::LOGTYPE_ACCESS);
+		if (!file_exists(dirname(__FILE__).'/'.self::INSTALL_SQL_FILE))
+			return (false);
+		else if (!$sql = version_compare(_PS_VERSION_, '1.4.0.10', '>=') ? Tools::file_get_contents(dirname(__FILE__).'/'.self::INSTALL_SQL_FILE) : file_get_contents(dirname(__FILE__).'/'.self::INSTALL_SQL_FILE))
+			return (false);
 
-		// Update table with new fields if not existing yet
-		$this->log('INSTALLATION - checking for field "comments" inside table "shopgate_order"', ShopgateLogger::LOGTYPE_DEBUG);
-		$db->Execute('SHOW COLUMNS FROM `'._DB_PREFIX_.'shopgate_order` LIKE \'comments\';');
-		if (!$db->NumRows())
-		{
-			$this->log('INSTALLATION - creating field "comments" inside table "shopgate_order"', ShopgateLogger::LOGTYPE_DEBUG);
-			if ($db->Execute('ALTER TABLE `'._DB_PREFIX_.'shopgate_order` ADD `comments` text NULL DEFAULT NULL AFTER `shipping_cost`;') === false)
-			{
-				$this->log('installation failed: unable to add field "comments" to table "shopgate_order". MySQL says: '.var_export($db->getMsgError(), true), ShopgateLogger::LOGTYPE_ERROR);
-				return false;
-			}
-		}
+		$sql = str_replace('PREFIX_', _DB_PREFIX_, $sql);
+		$sql = preg_split("/;\s*[\r\n]+/", $sql);
 
-		$this->log('INSTALLATION - checking for field "shop_number" inside table "shopgate_order"', ShopgateLogger::LOGTYPE_DEBUG);
-		$db->Execute('SHOW COLUMNS FROM `'._DB_PREFIX_.'shopgate_order` LIKE \'shop_number\';');
-		if (!$db->NumRows())
-		{
-			$this->log('INSTALLATION - creating field "shop_number" inside table "shopgate_order"', ShopgateLogger::LOGTYPE_DEBUG);
-			if ($db->Execute('ALTER TABLE `'._DB_PREFIX_.'shopgate_order` ADD `shop_number` varchar(16) NULL DEFAULT NULL AFTER `shipping_cost`;') === false)
-			{
-				$this->log('installation failed: unable to add field "shop_number" to table "shopgate_order". MySQL says: '.var_export($db->getMsgError(), true), ShopgateLogger::LOGTYPE_ERROR);
-				return false;
-			}
-		}
+		ShopGate::log('INSTALLATION - install tables', ShopgateLogger::LOGTYPE_DEBUG);
+
+		foreach ($sql as $query)
+			if ($query)
+				if (!$db::getInstance()->execute(trim($query)))
```

Comment:
Ah ok. Sorry didn't see this ;) You are right. I fixed it (locally) and will update the our module after you did the whole review. Could you please add a comment after you are done with the review?